Plugins/Public/Laz/Laws.cpp

```diff
@@ -0,0 +1,274 @@
+			ConPrint(L"LAWS: House Systems Loaded: %u \n", houseSystemsLoaded);
+			ConPrint(L"LAWS: Laws Loaded: %u \n", lawsLoaded);
+			ConPrint(L"LAWS: Contraband Items Loaded: %u \n", contrabandLoaded);
+		}
+	}
+
+	///////////////////////////////////////////////
+	//             Functions n'stuff             //
+	///////////////////////////////////////////////
+
+	void Laws::ContrabandWarning(uint iClientID, wstring wscText, ...) // WE WANT BOLD RED
+	{
+		wchar_t wszBuf[1024 * 8] = L"";
+		va_list marker;
+		va_start(marker, wscText);
+		_vsnwprintf(wszBuf, sizeof(wszBuf) - 1, wscText.c_str(), marker);
+
+		wstring wscXML = L"<TRA data=\"65281\" mask=\"-1\"/><TEXT>" + XMLText(wszBuf) + L"</TEXT>";
+		HkFMsg(iClientID, wscXML);
+	}
+
+	static void Laws::CheckCargo(int iClientID)
+	{
+		if (set_lawsContrabandEnabled)
+		{
+			uint iSystemID;
+			pub::Player::GetSystem(iClientID, iSystemID);
+
+			int iHoldSize;
+			list<CARGO_INFO> lstCargo;
+			HkEnumCargo((const wchar_t*)Players.GetActiveCharacterName(iClientID), lstCargo, iHoldSize);
+
+			for (map<uint, string>::iterator iter = mapHouseAndSystems.begin(); iter != mapHouseAndSystems.end(); iter++)
+			{
+
+				uint currentSector = iter->first;
+				string currentHouse = iter->second;
+
+				if (currentSector == iSystemID)
+				{
+					for (map<string, list<uint>>::iterator iter = mapHouseCargoList.begin(); iter != mapHouseCargoList.end(); iter++)
+					{
+						string currentHouseCompare = iter->first;
+						list<uint> cfgCargoLst = iter->second;
+
+						if (currentHouse == currentHouseCompare)
+						{
+							for (list<CARGO_INFO>::iterator i = lstCargo.begin(); i != lstCargo.end(); ++i)
+							{
+								uint correctCargo = i->iArchID;
+								foreach(cfgCargoLst, uint, cfgCargoIter)
+								{
+									if (*cfgCargoIter == correctCargo) // If their cargo is on the list
+									{
+										ContrabandWarning(iClientID, L"WARNING: You are carrying items which are illegal in this sector.");
+										return;
+									}
+								}
+							}
+						}
+					}
+					//PrintUserCmdText(iClientID, L"No apples were harmed in the making of this code.");
+				}
+			}
+		}
+	}
+
+	void Laws::PlayerLaunch(uint iShip, unsigned int iClientID)
+	{
+		pub::Player::GetSystem(iClientID, PrevSystemID);
+		CheckSector(iClientID);
+		CheckCargo(iClientID);
+	}
+
+	void Laws::JumpInComplete(unsigned int system, unsigned int ship)
+	{
+		uint iClientID = HkGetClientIDByShip(ship);
+		CheckSector(iClientID);
+		CheckCargo(iClientID);
+	}
+
+	// Checking System @ Jumpin
+	static void Laws::CheckSector(int iClientID) 
+	{
+		if (set_lawsJumpEnabled)
+		{
+			uint currentsystem;
+			pub::Player::GetSystem(iClientID, currentsystem);
+
+			for (map<uint, string>::iterator iter = mapHouseAndSystems.begin(); iter != mapHouseAndSystems.end(); iter++)
+			{
+				uint currentSystem = iter->first;
+				string currentHouse = iter->second;
+				if (currentSystem == currentsystem && currentSystem != PrevSystemID)
```

Comment:
yeah PrevSystemID could be for a totally different player. I'm assuming you only tested this with one player.
Also, naming two variables 'currentSystem' with different casing?